doc/blog/201906-porting-windows-calculator.md

```diff
@@ -0,0 +1,145 @@
+
+Build scripts are used for [iOS](https://github.com/nventive/calculator/blob/uno/src/CalcManager/build_ios.sh) and [WebAssembly](https://github.com/nventive/calculator/blob/uno/src/CalcManager/build.sh) to generate their native payloads, where as Windows and Android use their own VisualStudio projects for integration.
+
+Android's NDK also has its surprises, where the handling of exceptions is not enabled by default for NDKs below r19 for ARM32 ABIs. The calculation engine relies on the use of exceptions to handle cases such as divisions by zero, making the app crash in such conditions. Adding the `-lunwind` library to the linker solved that issue.
+
+### Challenge #2: Native code interop specifics
+
+There are multiple subtleties for building and using native code:
+- For iOS, it is [required to use the `__Internal`](https://github.com/nventive/calculator/blob/2657413f889ba26f2e3d78e82d384794fdad3aec/src/Calculator.Shared/CalcManager/CalculatorManager.Interop.cs#L16) known name, as native libraries are statically linked to main executable.
+- For Android, the integrated VS project does not automatically build for all target platforms (x86, armv7 and arm64), which means that [additional CI builds steps](https://github.com/nventive/calculator/blob/2657413f889ba26f2e3d78e82d384794fdad3aec/.vsts-ci.Windows.yml#L37) have to be executed so the proper binaries are used all at once when creating the final APK. Also, standard project dependencies cannot be used (because of the multi ABI builds), and the [`AndroidNativeLibrary`](https://github.com/nventive/calculator/blob/2657413f889ba26f2e3d78e82d384794fdad3aec/src/Calculator.Droid/Calculator.Droid.csproj#L101) items must be added in the project.
+- For WebAssembly, the module needs to be built using the `-s SIDE_MODULE=1` support, with additional support for C++ standard library, as it is [not automatically added by emscripten](https://github.com/emscripten-core/emscripten/wiki/Linking#system-libraries) for dynamic modules.
+
+### Challenge #3: Converting the C++/CX code to C#
+
+The rest of the application is using Microsoft's C++/CX to use WinRT APIs and particularly WinUI APIs. This makes for a very C#-like code, that can be converted to C# using a set of regular expressions that convert `::` to `.`, or `dynamic_cast<FrameworkElement>(fe)` to `((FrameworkElement)fe)`, and the rest of the WinRT API calls just match directly, as Uno provides the full APIs signatures of WinRT.
+
+That conversion converts C++/CX code to roughly 80% of C#-compatible code, with some manual adjustments to get fully functional C#. For instance, C++ iterators need to be converted to `foreach` loops, or string to numbers conversions using .NET built-in conversions.
```

Comment:
'converts C++/CX code to roughly 80% of C#-compatible code' - 'converts roughly 80% of C++/CX code to C#-compatible code'?